extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -183,7 +217,111 @@ export class ResourceTypePickerDialog extends DialogBase {
 		this._dialogObject.content = [tab];
 	}
 
+	private createResouceButtonsContainer(): azdata.ListViewComponent {
+		const tags = this.getAllResourceTags(this._resourceTypes);
+		if (!tags.includes('All')) {
+			tags.splice(0, 0, 'All');
+		}
+		const items: azdata.ListViewOption[] = [];
+		tags.forEach((t: string) => {
+			items.push({
+				label: t,
+				id: t
+			});
+		});
+		const listView = this._view.modelBuilder.listView().withProps({
+			title: {
+				text: 'Categories',
+			},
+			CSSStyles: {
+				'width': '140px'
+			},
+			options: items,
+			selectedOptionId: items[0].id
+		}).component();
+		this._toDispose.push(listView.onDidClick((e) => {
+			this.filterResources();
+		}));
+
+		return listView;
+	}
+
+	private filterResources(): void {
+
+		let search = this._resourceSearchBox.value;
+		if (search) {
+			search = search.toLowerCase();
+		} else {
+			search = '';
+		}
+		const tag = this._resourceTagsListView.selectedOptionId!;
+
+		let filteredResourceTypes: ResourceType[] = [];
+
+		if (tag !== 'All') {
+			this._resourceTypes.forEach(element => {
+				if (element.tags?.includes(tag)) {
+					filteredResourceTypes.push(element);
+				}
+			});
+		} else {
+			filteredResourceTypes = this._resourceTypes;
+		}
+
+		let filteredResourceTypesonSearch: ResourceType[] = [];
+		filteredResourceTypes.forEach(element => {
+			if (element.displayName.toLowerCase().includes(search!)) {
+				filteredResourceTypesonSearch.push(element);
+			}
+		});
+		filteredResourceTypes.forEach(element => {
+			if (!element.displayName.toLowerCase().includes(search!) && element.description.toLowerCase().includes(search!)) {
+				filteredResourceTypesonSearch.push(element);
+			}
+		});
+
+		const cards = filteredResourceTypesonSearch.map((resourceType) => {
+			return <azdata.RadioCard>{
```

Comment:
Instead of constantly creating new objects why not just keep a map of all the cards and then just select the ones from that map?